File: C++/523. Continuous Subarray Sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        int psum=0;
        unordered_map<int,int>mp;
        int n=nums.size();        
        for(int i=0;i<n;i++)
        {
            psum+=nums[i];
            psum%=k;
            int s=i+1;
            if(s>1 && psum==0)
            return true;
            if(mp.find(psum)!=mp.end())
            {
                int j=mp[psum];
                if(i-j>1 )
                return true;
            }
            else
            mp[psum]=i;
        }
        return false;
    }
};

File: JAVA/523. Continuous Subarray Sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) 
    {
      int n=nums.length;
      int sum=0;
    Map<Integer,Integer> map=new HashMap<>();
    map.put(0,-1);
    for(int i=0; i<n; i++)
    {
        sum=sum+nums[i];
      if(map.containsKey(sum%k))
      {
        int index=map.get(sum%k);
        if(i-index>=2)
        {
            return true;
        }
      }
      else
      {
        map.put(sum%k,i);
      }
    }
    return false;
    }
}
